### PR TITLE
JS validation for the frontpage

### PR DESCRIPTION
**Pull request by HrBjarup**

### CODE REVIEW

**Comment by MalteMagnussen** [comment]
I'll try to recreate the issue locally and see what the issue is.

**Comment by MalteMagnussen** [comment]
image
Works fine on "ChangeOrderCarport" atm.

**Comment by Runi-VN** [comment]
Does not work for me on "ChangeOrderCarport"
image

**Comment by MalteMagnussen** [comment]
https://github.com/HrBjarup/Fog-Carport/pull/175/commits/a923b22e96b37fd15ce6d697d270f9ddce100e92 - Fix attempt. In orderCarport.java this is the line causing the issue: 
image

**Comment by MalteMagnussen** [comment]
Could you guys test it again? I can't reproduce the bug.

File: FogCarport/src/main/java/presentation/FrontController.java [hunk 1]
```diff
@@ -34,15 +34,18 @@ protected void processRequest(HttpServletRequest request, HttpServletResponse re
...
+            }
             Command action = Command.from(request);
             String target = action.execute(request, logic);
             request.setAttribute("target", target);
             request.getRequestDispatcher("index.jsp").forward(request, response);
 
         } catch (UserException | DataException | AlgorithmException ex) // AlgorithmException should redirect user somewhere away from SVG and partslist but keep session
         {
-            request.setAttribute("target", "homepage");
+            request.setAttribute("target", "login");
```
Comment (MalteMagnussen) [inline FogCarport/src/main/java/presentation/FrontController.java]: This is "errorMessage" now, which is a customized error message page.

Comment (MalteMagnussen) [inline FogCarport/src/main/java/presentation/FrontController.java]:
```suggestion
            request.setAttribute("target", "errorMessage");
```

**Comment by Runi-VN** [comment]
As far as I can see, it works as it should!! 😄 

I noticed one thing - 
We validate on the customer first, so if the user is validated correctly, but the order is not, the following happens:
1) Customer account is created
2) Order is **not** created
3) Customer is sent to error page, and is notified of the error in the order form, but **not** that an user was created.

This really is a *edge case* if I know one, and here at 2am I can't really think straight, but could we mix up the validation, and validate on order first?

**Comment by MalteMagnussen** [comment]
> As far as I can see, it works as it should!! 😄
> 
> I noticed one thing -
> We validate on the customer first, so if the user is validated correctly, but the order is not, the following happens:
> 
> 1. Customer account is created
> 2. Order is **not** created
> 3. Customer is sent to error page, and is notified of the error in the order form, but **not** that an user was created.
> 
> Steps to re-create:
> Fill out the form, tick the box for 'shed' but do not enter any values.
> Fill out customer info and hit submit.
> 
> This really is an _edge case_ if I know one, and here at 2am I can't really think straight, but could we mix up the validation, and validate on order first?

The solution to this is "transactions" afaik https://docs.oracle.com/javase/tutorial/jdbc/basics/transactions.html

**Pull request merged**